From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Oleksii Kulinich <entaytion@gmail.com>
Date: Sat, 15 Aug 2026 00:28:31 +0300
Subject: Remove duplicate R8 keep rule

---
 TMessagesProj/proguard-rules.pro | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/TMessagesProj/proguard-rules.pro b/TMessagesProj/proguard-rules.pro
index 0000000..0000000 100644
--- a/TMessagesProj/proguard-rules.pro
+++ b/TMessagesProj/proguard-rules.pro
@@ -15,7 +15,6 @@
 -keep class org.telegram.messenger.camera.* { *; }
 -keep class org.telegram.messenger.secretmedia.* { *; }
 -keep class org.telegram.messenger.support.* { *; }
--keep class org.telegram.messenger.support.* { *; }
 -keep class org.telegram.messenger.time.* { *; }
 -keep class org.telegram.messenger.video.* { *; }
 -keep class org.telegram.messenger.voip.* { *; }
@@ -131,4 +130,4 @@
 
 # Use -keep to explicitly keep any other classes shrinking would remove
 -dontoptimize
--dontobfuscate
\ No newline at end of file
+-dontobfuscate
